sdk/lib/opentelemetry/sdk/internal.rb

```diff
@@ -20,8 +20,12 @@ def valid_key?(key)
         key.instance_of?(String)
       end
 
+      def valid_numeric?(value)
+        value.is_a?(Integer) || value.is_a?(Float)
```

Comment:
Two things:
1. I wonder if we should explicitly use `.instance_of?` to restrict values to instances of these specific classes, in case someone tries to subclass them & that 💥 in `protobuf` serialization later.
2. What's the current state of `BigInteger`? IIRC it was merged with `Integer` some releases of MRI ago. I'm not sure whether we should check for it here or if we can now comfortably assume `value.is_a?(Integer)` covers that.

Reply:
100% agree with moving to `instance_of?` - that's a great idea.

Reply:
I think `Bignum` and `Fixnum` are kaput as of 2.4.0: https://github.com/ruby/ruby/blob/master/doc/ChangeLog-2.4.0#L4927-L4963

Reply:
changed from `is_a?` to `instance_of?` here 0c1f281cb7980f63e037a50dd809f4fdf627b1ed